pkg/auth/dependency/webapp/authenticate_provider_impl.go

```diff
@@ -483,7 +492,7 @@ func (p *AuthenticateProviderImpl) LinkIdentityProvider(w http.ResponseWriter, r
 	q.Set("redirect_uri", r.URL.Path)
 	webappSSOState.SetRequestQuery(q.Encode())
 	state := sso.State{
-		Action: "link",
+		Action: action,
```

Comment:
I think we need to fix the `userID` if it is promotion? The `userID` in this function is from `auth.GetSession`

Reply:
Indeed, I didn't saw it when rebasing.